pkg/actor/validate_version.go

```diff
@@ -293,15 +300,9 @@ func (v *versionChecker) Act(ctx context.Context, cluster *resource.Cluster) err
 		return err
 	}
 
-	dp := metav1.DeletePropagationForeground
-
-	//delete the job only if we have managed to get the version and we do not have any errors
-	err = clientset.BatchV1().Jobs(cluster.Namespace()).Delete(ctx, job.Name, metav1.DeleteOptions{
-		GracePeriodSeconds: ptr.Int64(5),
-		PropagationPolicy:  &dp,
-	})
-	if err != nil {
-		log.Error(err, "failed to delete the job")
+	// If we got here, the version checker job was successful. Delete it.
+	if dErr := deleteJob(ctx, cluster, clientset, job); dErr != nil {
+		log.Error(dErr, "failed to delete the job")
```

Comment:
Does log.Error return an error? Wondering if we should return here

Reply:
log.Error doesn't return an error, but do you think returning here is the right thing to do? I could see an argument either way - let the version checker finish and potentially have a dangling job, or just return and say the version checker failed.

Reply:
If we error here I would worry that we can't move forward with the next step. Having a dangling job is probably ok IMO, better than having version checker come back as ok but fail on clean up. Maybe add more context in the error message. Check passed but delete failed.

Reply:
I've updated the error message as we just discussed. This PR should be good to go.